examples/ex18.hpp

```diff
@@ -1,490 +1,279 @@
 //                  MFEM Example 18 - Serial/Parallel Shared Code
-
+//                      (Implementation of Time-dependent DG Operator)
+//
+// This code provide example problems for the Euler equations and implements
+// the time-dependent DG operator given by the equation:
+//
+//            (u_t, v)_T - (F(u), ∇ v)_T + <F̂(u, n), [[v]]>_F = 0.
+//
+// This operator is designed for explicit time stepping methods. Specifically, the
+// function DGHyperbolicConservationLaws::Mult implements the following transformation:
+//
+//                             u ↦ M⁻¹(-DF(u) + NF(u))
+//
+// where M is the mass matrix, DF is the weak divergence of flux, and NF is the
+// interface flux. The inverse of the mass matrix is computed element-wise by
+// leveraging the block-diagonal structure of the DG mass matrix. Additionally,
+// the flux-related terms are computed using the HyperbolicFormIntegrator.
+//
+// The maximum characteristic speed is determined for each time step. For more details,
+// refer to the documentation of DGHyperbolicConservationLaws::Mult.
+//
+
+#include <functional>
 #include "mfem.hpp"
 
-using namespace std;
-using namespace mfem;
-
-// Problem definition
-extern int problem;
-
-// Maximum characteristic speed (updated by integrators)
-extern double max_char_speed;
-
-extern const int num_equation;
-extern const double specific_heat_ratio;
-extern const double gas_constant;
+namespace mfem
+{
 
-// Time-dependent operator for the right-hand side of the ODE representing the
-// DG weak form.
-class FE_Evolution : public TimeDependentOperator
+/// @brief Time dependent DG operator for hyperbolic conservation laws
+class DGHyperbolicConservationLaws : public TimeDependentOperator
 {
 private:
-   const int dim;
-
-   FiniteElementSpace &vfes;
-   Operator &A;
-   SparseMatrix &Aflux;
-   DenseTensor Me_inv;
-
-   mutable Vector state;
-   mutable DenseMatrix f;
-   mutable DenseTensor flux;
+   const int dim; // domain dimension
```

Comment:
It looks like this member variable is unused?

Reply:
I just removed it. Thank you!